### PR TITLE
feat(react): add rspack module federation support

### PR DESCRIPTION
**Pull request by Coly010**
- feat(react): add remote rspack module federation support
- feat(react): add host rspack module federation support
- feat(react): add federate module rspack module federation support
- fix(react): migration test

## Current Behavior

We do not have an option to generate a react host and remote with rspack

## Expected Behavior

Add rspack as an option when generating host and remote

## Related Issue(s)

Fixes #

### CODE REVIEW

**Review by isaacplmann — APPROVED** [review]

**Review by jaysoo — APPROVED** [review]

File: packages/react/src/utils/maybe-js.ts [hunk 1]
```diff
@@ -1,5 +1,8 @@
-export function maybeJs(options: { js?: boolean }, path: string): string {
+export function maybeJs(
+  options: { js?: boolean; useJsx?: boolean },
+  path: string
+): string {
   return options.js && (path.endsWith('.ts') || path.endsWith('.tsx'))
-    ? path.replace(/\.tsx?$/, '.js')
+    ? path.replace(/\.tsx?$/, options.useJsx ? '.jsx' : '.js')
```
[the remark this answers is not in the document]
Comment (Coly010) [reply packages/react/src/utils/maybe-js.ts:6]: yes :)

File: e2e/react/src/react-module-federation.test.ts [hunk 2]
```diff
@@ -19,13 +19,13 @@ import {
...
 
-describe('React Module Federation', () => {
+describe('debugthis React Module Federation', () => {
   describe('Default Configuration', () => {
     beforeAll(() => {
       newProject({ packages: ['@nx/react', '@nx/webpack'] });
     });
 
-    afterAll(() => cleanupProject());
+    // afterAll(() => cleanupProject());
```
Comment (AgentEnder) [inline e2e/react/src/react-module-federation.test.ts:28]: Revert these

**Review by AgentEnder — CHANGES_REQUESTED** [review]

File: e2e/react/src/react-module-federation.test.ts (hunk 2 above)
Comment (Coly010) [reply e2e/react/src/react-module-federation.test.ts:28]: ah good catch

File: docs/generated/packages/react/generators/setup-ssr.json [hunk 3]
```diff
@@ -39,6 +39,12 @@
         "hidden": true,
         "description": "Extra include entries in tsconfig.",
         "default": []
+      },
+      "bundler": {
+        "description": "The bundler to use.",
+        "type": "string",
+        "enum": ["rspack", "webpack"],
+        "default": "webpack"
```
Comment (zmzlois) [inline docs/generated/packages/react/generators/setup-ssr.json:47]: should this be Rspack?

File: e2e/react/src/react-module-federation.test.ts [hunk 4]
```diff
@@ -40,7 +40,7 @@ describe('React Module Federation', () => {
         const remote3 = uniq('remote3');
 
         runCLI(
-          `generate @nx/react:host ${shell} --remotes=${remote1},${remote2},${remote3} --e2eTestRunner=cypress --style=css --no-interactive --skipFormat --js=${js}`
+          `generate @nx/react:host ${shell} --remotes=${remote1},${remote2},${remote3} --bundler=webpack --e2eTestRunner=cypress --style=css --no-interactive --skipFormat --js=${js}`
```
Comment (zmzlois) [inline e2e/react/src/react-module-federation.test.ts:43]: should test in the scenario in rspack too?

**Review by AgentEnder — APPROVED** [review]

File: e2e/react/src/react-module-federation.test.ts (hunk 4 above)
Comment (Coly010) [reply e2e/react/src/react-module-federation.test.ts:43]: there's a separate test file testing rspack

File: docs/generated/packages/react/generators/setup-ssr.json (hunk 3 above)
Comment (Coly010) [reply docs/generated/packages/react/generators/setup-ssr.json:47]: no, setup-ssr can still be used in non-module federation applications. only module-federation related generators have been updated to default to rspack

**Pull request merged**